masakari-controller/db/api.py

```diff
@@ -0,0 +1,524 @@
+                notification_id, recover_to, recover_by):
+    # INSERT INTO vm_list ( create_at, deleted, uuid, progress, retry_cnt,
+    #   notification_id, recover_to, recover_by ) VALUES ( ... )
+    with _sqlalchemy_error():
+        vm_list = VmList(create_at=create_at, deleted=deleted, uuid=uuid,
+                         progress=progress, retry_cnt=retry_cnt,
+                         notification_id=notification_id,
+                         recover_to=recover_to,
+                         recover_by=recover_by)
+    session.add(vm_list)
+    session.commit()
+    return vm_list
+
+
+@_session_handle
+def get_all_reserve_list_by_hostname_not_deleted(session, hostname):
+    # SELECT * FROM reserve_list WHERE deleted=0 AND hostname=:hostname
+    with _sqlalchemy_error():
+        res = session.query(ReserveList).filter_by(hostname=hostname).\
+            filter_by(deleted=0).all()
+    return res
+
+
+@_session_handle
+def get_one_reserve_list_by_cluster_port_for_update(session, cluster_port,
+                                                    notification_hostname):
+    # SELECT id,hostname FROM reserve_list
+    #   WHERE deleted=0 and cluster_port=:cluster_port
+    #   and hostname!=:notification_hostname
+    #   ORDER by create_at asc limit 1 FOR UPDATE
+    with _sqlalchemy_error():
+        res = session.query(
+            ReserveList).with_for_update().filter_by(deleted=0).filter_by(
+                cluster_port=cluster_port).filter(
+                    ReserveList.hostname != notification_hostname).order_by(
+                        asc(ReserveList.create_at)).first()
+    return res
+
+
+@_retry_on_deadlock
+@_session_handle
+def update_reserve_list_by_hostname_as_deleted(session, hostname, delete_at):
+    # UPDATE reserve_list SET deleted=1, delete_at=:delete_at
+    #   WHERE hostname=:hostname
+    res = session.query(ReserveList).filter_by(hostname=hostname).\
+        update({'delete_at': delete_at, 'deleted': 1})
+    return res
+
+
+@_retry_on_deadlock
+@_session_handle
+def update_reserve_list_by_cluster_port_as_deleted(session, delete_at,
+                                                   cluster_port):
+    # UPDATE reserve_list SET deleted=1, delete_at=:delete_at
+    #   WHERE cluster_port=:cluster_port
+    res = session.query(ReserveList).filter_by(cluster_port=cluster_port).\
+        update({'delete_at': delete_at, 'deleted': 1})
+    return res
+
+
+@_session_handle
+def get_old_records_notification(session, border_time):
+    # sql = "SELECT id FROM notification_list " \
+    #       "WHERE progress = 0 AND create_at < '%s'" \
+    #       % (border_time_str)
+    with _sqlalchemy_error():
+        res = session.query(NotificationList).filter(
+            NotificationList.progress == 0,
+            NotificationList.create_at < border_time).all()
+    return res
+
+
+@_retry_on_deadlock
+@_session_handle
+def delet_expired_notification(session, progress, update_at, delete_at, id):
```

Comment:
IMO, the method don't need to have "progress" arg since following query don't use it.